script/deploy-core.s.sol

```diff
@@ -45,44 +65,92 @@ contract CoreDeployer is Script {
-
-            vm.broadcast(deployer);
-            LBPair pairImplementation = new LBPair(factoryV2);
-            console.log("LBPair implementation deployed -->", address(pairImplementation));
-
-            vm.broadcast(deployer);
-            LBRouter routerV2 = new LBRouter(
-                factoryV2,
-                ISovrynLBFactoryV1(deployment.factoryV1),
-                IWNATIVE(deployment.wNative)
-            );
-            console.log("LBRouter deployed -->", address(routerV2));
-
-            vm.startBroadcast(deployer);
-            LBQuoter quoter = new LBQuoter(
-                deployment.factoryV1,
-                address(factoryV2),
-                address(routerV2)
-            );
-            console.log("LBQuoter deployed -->", address(quoter));
-
-            factoryV2.setLBPairImplementation(address(pairImplementation));
-            console.log("LBPair implementation set on factoryV2\n");
-
-            uint256 quoteAssets = ILBLegacyFactory(deployment.factoryV2).getNumberOfQuoteAssets();
-            for (uint256 j = 0; j < quoteAssets; j++) {
-                IERC20 quoteAsset = ILBLegacyFactory(deployment.factoryV2).getQuoteAsset(j);
-                factoryV2.addQuoteAsset(quoteAsset);
+            if(deployment.factoryV2 == address(0)) {
+                console.log("Deploying factory v2...");
+                factoryV2 = Upgrades.deployTransparentProxy(
+                    "LBFactory.sol",
+                    deployment.owner,
+                    ""
+                );
+
+                console.log("LBFactory deployed -->", factoryV2);
+            } else {
+                factoryV2 = deployment.factoryV2;
+            }
+
+            deployment.factoryV2 = factoryV2;
+
+            if(deployment.lbPairImplementation == address(0)) {
+                console.log("Deploying lbPairImplementation...");
+                lbPairImplementation = address(new LBPair(ILBFactory(factoryV2)));
+                console.log("lbPairImplementation deployed -->", lbPairImplementation);
+            } else {
+                lbPairImplementation = deployment.lbPairImplementation;
+            }
+
+            deployment.lbPairImplementation = lbPairImplementation;
+
+            
+            if(deployment.lbPairUpgradeableBeacon == address(0)) {
+                console.log("Deploying lbPairUpgradeableBeacon...");
+                lbPairUpgradeableBeacon = address(new LBPairUpgradeableBeacon(lbPairImplementation, deployer, factoryV2));
+                console.log("lbPairUpgradeableBeacon deployed -->", lbPairUpgradeableBeacon);
+            } else {
+                lbPairUpgradeableBeacon = deployment.lbPairUpgradeableBeacon;
+            }
+
+            deployment.lbPairUpgradeableBeacon = lbPairUpgradeableBeacon;
+
+            console.log("initializing lbFactory...");
+            LBFactory(factoryV2).initialize(deployer, deployer, FLASHLOAN_FEE, lbPairUpgradeableBeacon);
```

Comment:
the first param is the feeRecipient - no reason to set it to the deployer   
pls add feeRecipient deployment param  
on the testnet it should be the multisig initially, then changed to the feeSharingCollector, on the mainnet - feeSharingCollector (needs to be adjusted - i created a separate ticket for this).  
```suggestion
            LBFactory(factoryV2).initialize(deployment.feeRecipient, deployer, FLASHLOAN_FEE, lbPairUpgradeableBeacon);
```

Reply:
Updated in this [commit](https://github.com/DistributedCollective/sovryn-lb-dex/pull/3/commits/e7e76f053b2778191a3461452b40d2f0dfe647e7)